src/prefect/blocks/abstract.py

```diff
@@ -82,6 +101,20 @@ async def notify(self, body: str, subject: Optional[str] = None) -> None:
             subject: The subject of the notification.
         """
 
+    _raise_on_failure: bool = False
+
+    @contextmanager
+    def raise_on_failure(self) -> Generator[None, None, None]:
```

Comment:
I think this might be better suited on the `AbstractAppriseNotificationBlock` class or the logic for using this contextmanager in `notify` needs to be moved to `NotificationBlock`. I think it's a bit misleading in its current implementation because using this contextmanager only changes behavior if a notification block has been updated to use it.

Reply:
My plan was to go through the other collections blocks and make sure they do something sensible when they get errors, but it's a chicken & egg kinda situation.  As long as the functions raise _something_ we're good.  `apprise` was just a bit wonky in that it silently failed in this very imperceptible way.

Reply:
Gotcha, that makes sense. The large majority of notification blocks that I'm aware of are in the core library, so this should cover most cases.